src/inapp/utils.ts

```diff
@@ -114,6 +115,103 @@ export const sortInAppMessages = (messages: Partial<InAppMessage>[] = []) => {
   return messages.sort(by(['priorityLevel', 'asc'], ['createdAt', 'asc']));
 };
 
+/**
+ * detect amount of local storage remaining (quota) and used (usage).
+ * if usageDetails exist (not supported in Safari), use this instead of usage.
+ */
+export const determineRemainingStorageQuota = async () => {
```

Comment:
@ts-nguyen @mprew97 I'm not quite happy with the implementation I've done for either of these utility methods because they feel a bit clunky and convoluted. Would love some feedback on these in particular